melodies_monet/driver.py

```diff
@@ -1565,6 +1588,142 @@ def plotting(self):
+
+                        
+                            # Resample model data to target pressures using stratify
+                            da_wrf_const = resample_stratify(ds_model[modvar], target_pressures, ds_model['pressure_model'], axis=1, interpolation='linear', extrapolation='nan')
+                            da_wrf_const.name = modvar
+                        
+                            # Create target_pressures DataArray
+                            da_target_pressures = xr.DataArray(target_pressures, dims=('z'))
+                            da_target_pressures.name = 'target_pressures'
+                        
+                            # Merge DataArrays into a single Dataset
+                            ds_wrf_const = xr.merge([da_wrf_const, da_target_pressures])
+                            ds_wrf_const = ds_wrf_const.set_coords('target_pressures')
+                        
+                            # Debugging: print merged dataset for model curtain
+                            ##print(ds_wrf_const)
+                        
+                            # Ensure model_data_2d is properly reshaped for the contourfill plot
+                            model_data_2d = ds_wrf_const[modvar].squeeze()
+                        
+                            # Debugging: print reshaped model data shape
+                            ##print(f"Reshaped model data shape: {model_data_2d.shape}")
+                            
+                            #### model_data_2d for curtain plot ready ####
+
+
+                            # Fetch model pressure and other model and observation data from "pairdf" (for scatter plot overlay)
+                            time = pairdf['time']
+                            obs_pressure = pairdf['pressure_obs']  
+                            ##print(f"Length of time: {len(time)}") #Debugging
+                            ##print(f"Length of obs_pressure: {len(obs_pressure)}") #Debugging
+                        
+                            # Generate the curtain plot using airplots.make_curtain_plot
+                            try:
+                                outname_pair = f"{outname}_{obs_label}_vs_{model_label}.png"
+
+                                print(f"Saving curtain plot to {outname_pair}...")
+                        
+                                ax = airplots.make_curtain_plot(
+                                    time=pd.to_datetime(time),
+                                    altitude=target_pressures,  # Use target_pressures for interpolation
+                                    model_data_2d=model_data_2d,  # Already reshaped to match the expected shape
+                                    pairdf=pairdf,  #use pairdf for scatter overlay (model and obs)
+                                    obs_pressure=obs_pressure,  # Pressure_obs for obs scatter plot
```

Comment:
obs_pressure and pairdf are in a different order here than in the make_curtain_plot routine in the aircraftplots.py file and several of the ones below are also in a different order. Can you make this consistent between these two files? It makes it easier keep things consistent in the future.